### DISCUSSION TITLE
Support passing values to `select(fields...)` for `insert into ... select from`

### DISCUSSION THREAD

**Discussion by SadiinsoSnowfall** [diesel-rs/diesel#3068] [Q&A] [answered]
The issue #1106 and then the PR #1496 added support for this kind of request : 
```sql
INSERT INTO the_table (id, name, description)
    SELECT id + 1, name, description FROM the_table
```
Using this construct : 
```rust
table! (
    the_table {
        id -> Integer,
        name -> Text,
        description -> Text,
    }
);

diesel::insert_into(the_table)
    .values(the_table.select(id + 1, name, description))
    .execute(conn);
```
However, it is not possible to re-create this following query using Diesel:
```sql
INSERT INTO the_table (id, name, description)
    SELECT 55, name, description FROM the_table
```
```rust
diesel::insert_into(the_table)
    .values(the_table.select(55, name, description))
    .execute(conn);
```
This rust code produce the following errors:
```
the trait bound `{integer}: diesel::Expression` is not satisfied
the trait bound `{integer}: SelectableExpression<the_table::table>` is not satisfied
the trait bound `{integer}: NonAggregate` is not satisfied
```

Would it be possible to support passing direct values to `select(...)` in order to support this kind of SQL query ?

**Comment by weiznich** [comment, marked as answer]
I've converted this to a discussion as this clearly is not a bug report and it does not follow the issue template (that's mandatory).
To answer this as question: Yes this is already possible. You need to convert the rust value explicitly to a diesel query builder expression. This can be done by for example calling [`IntoSql::into_sql()`](https://docs.diesel.rs/1.4.x/diesel/expression/trait.IntoSql.html#method.into_sql).
